src/main/kotlin/tf/sou/mc/pal/listeners/ChestListener.kt

```diff
@@ -108,8 +110,12 @@ class ChestListener(private val pal: ChestPal) : Listener {
         }
         // Clean up.
         badItems.forEach {
-            inventory.remove(it)
-            event.player.inventory.addItem(it)
+            if (it != null) {
+                inventory.remove(it)
+            }
+            if (it != null) {
+                event.player.inventory.addItem(it)
+            }
```

Comment:
Please remove. Two things: 1. You are validating the same condition twice here 2. It's best to keep the call site clean. `findBadItems` should return `List<ItemStack>` instead of `List<ItemStack?>`. More info down below :)